src/main/java/org/jenkinsci/plugins/jiraext/view/Transition.java

```diff
@@ -50,27 +50,48 @@ public Transition(String transitionName)
 
     @Override
     public void perform(List<JiraCommit> jiraCommitList,
-                        AbstractBuild build, Launcher launcher, BuildListener listener)
-    {
-        try
-        {
-            for (JiraCommit jiraCommit : JiraCommit.filterDuplicateIssues(jiraCommitList))
-            {
+                        AbstractBuild build, Launcher launcher, BuildListener listener) {
+        try {
+            for (JiraCommit jiraCommit : JiraCommit.filterDuplicateIssues(jiraCommitList)) {
                 listener.getLogger().println("Transition a ticket: " + jiraCommit.getJiraTicket());
-                listener.getLogger().println("transitionName: " + transitionName);
-                try
-                {
-                    getJiraClientSvc().changeWorkflowOfTicket(jiraCommit.getJiraTicket(), transitionName);
+
+                //Make a copy of all transitions since we will be removing the ones that are done
+                List<String> transitions = Arrays.stream(StringUtils.split(transitionName, ","))
+                        .map(t -> t.trim())
+                        .collect(Collectors.toList());
+                boolean didAnyTransition = false;
+
+                //Loop through all transitions until one works
+                while (transitions.size() > 0) {
+                    Iterator<String> transitionIter = transitions.iterator();
+                    boolean didTransition = false;
+                    while (transitionIter.hasNext()) {
+                        String transition = transitionIter.next();
+                        try {
+                            getJiraClientSvc().changeWorkflowOfTicket(jiraCommit.getJiraTicket(), transition);
+
+                            //Transition worked. Remove it from the list, and try the next one
+                            listener.getLogger().println("Performed transition: " + transition);
+                            transitionIter.remove();
+                            didAnyTransition = true;
+                            didTransition = true;
+                        } catch (Throwable t) {
+                            //Ignore and try next transition
```

Comment:
It is probably worth logging that this transition was tried but there was some exception, which we are assuming to mean that the transition itself was not valid vs some other type of error. This at least makes visible (and thus usable) on what actions are happening -- the only way to see this currently is looking at source code.

Additionally - we need the ability to see this exception under trace / debug logging to facilitate debugging. It's possible that we are hitting some other error - we do not discriminate here, everything is caught